core/src/main/java/org/elasticsearch/action/admin/indices/validate/query/QueryExplanation.java

```diff
@@ -65,6 +76,9 @@ public String getExplanation() {
     @Override
     public void readFrom(StreamInput in) throws IOException {
         index = in.readString();
+        if (in.getVersion().onOrAfter(Version.V_5_4_0_UNRELEASED)) {
+            shard = in.readInt();
+        }
```

Comment:
do you want to set it to `-1` or something in the `else` arm so it is obviously bogus?